sklearn/cluster/tests/test_hierarchical.py

```diff
@@ -140,6 +141,30 @@ def test_agglomerative_clustering_wrong_arg_memory():
     assert_raises(ValueError, clustering.fit, X)
 
 
+class Dummy(object):
```

Comment:
Don't call this Dummy. It is not straightforward for something that it is not used to the code. `DummyMemory` is more informative